### ISSUE TITLE
MetaPruner can not work with ViTForImageClassification when setting   iterative_steps=5 or larger steps

### ISSUE THREAD

**Issue by Aesop-programmer** [VainF/Torch-Pruning#411]
```
import torch_pruning as tp
from transformers.models.vit.modeling_vit import ViTSelfAttention, ViTSelfOutput
model = ViTForImageClassification.from_pretrained('google/vit-base-patch16-224-in21k',
                                                              num_labels=100,
                                                              id2label=id2label,
                                                              label2id=label2id).eval().to("cuda")
example_inputs = torch.randn(1,3,224,224).to("cuda")

num_heads = {}
ignored_layers = [model.classifier]
for m in model.modules():
    if isinstance(m, ViTSelfAttention):
        num_heads[m.query] = m.num_attention_heads
        num_heads[m.key] = m.num_attention_heads
        num_heads[m.value] = m.num_attention_heads
imp = tp.importance.GroupNormImportance(2)
pruner = tp.pruner.MetaPruner(
                model, 
                example_inputs,
                iterative_steps=5,
                global_pruning=False, # If False, a uniform pruning ratio will be assigned to different layers.
                importance=imp, # importance criterion for parameter selection
           
                ignored_layers=ignored_layers,

                num_heads = num_heads,
                prune_head_dims=False,
                prune_num_heads=True,
                head_pruning_ratio=(1.0/3.0), # disabled when prune_num_heads=False
                round_to=2,
)
for i, g in enumerate(pruner.step(interactive=True)):
    g.prune()

for m in model.modules():
    if isinstance(m, ViTSelfAttention):
        print("num_heads:", m.num_attention_heads, 'head_dims:', m.attention_head_size, 'all_head_size:', m.all_head_size, '=>')
        m.num_attention_heads = pruner.num_heads[m.query]
        m.attention_head_size = m.query.out_features // m.num_attention_heads
        m.all_head_size = m.query.out_features
        print("num_heads:", m.num_attention_heads, 'head_dims:', m.attention_head_size, 'all_head_size:', m.all_head_size)
        print()
```
After running the above code, it will show
**RuntimeError: shape '[12, -1]' is invalid for input of size 704**

I wonder whether  **dim_imp = imp.view(ch_groups, -1).mean(dim=0)** in line 420 in metapruner.py can be a problem, becasue the head dim had been 11 not 12. 

Is my setting wrong or is it a bug ?

**Comment by VainF** [comment]
Hi this bug was fixed in the latest commit! Please update to v1.4.2 to address this.

**Comment by Aesop-programmer** [comment]
It works ! Thank you very much.
